WooCommerce/Classes/ViewRelated/Products/Edit Product/Shipping Settings/ProductShippingSettingsViewController.swift

```diff
@@ -0,0 +1,236 @@
+import UIKit
+import Yosemite
+
+// MARK: - ProductShippingSettingsViewController
+//
+final class ProductShippingSettingsViewController: UIViewController {
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    // Editable data
+    //
+    private var weight: String?
+    private var length: String?
+    private var width: String?
+    private var height: String?
+    private var shippingClassSlug: String?
+
+    /// Table Sections to be rendered
+    ///
+    private let sections: [Section] = [
+        Section(rows: [.weight, .length, .width, .height]),
+        Section(rows: [.shippingClass])
+    ]
+
+    private let product: Product
+    private let shippingSettingsService: ShippingSettingsService
+
+    init(product: Product,
+         shippingSettingsService: ShippingSettingsService = ServiceLocator.shippingSettingsService) {
+        self.product = product
+        self.shippingSettingsService = shippingSettingsService
+
+        self.weight = product.weight
+        self.length = product.dimensions.length
+        self.width = product.dimensions.width
+        self.height = product.dimensions.height
+        self.shippingClassSlug = product.shippingClass
+
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureNavigationBar()
+        configureMainView()
+        configureTableView()
+    }
+}
+
+// MARK: - View Configuration
+//
+private extension ProductShippingSettingsViewController {
+
+    func configureNavigationBar() {
+        title = NSLocalizedString("Shipping", comment: "Product Shipping Settings navigation title")
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(completeUpdating))
+    }
+
+    func configureMainView() {
+        view.backgroundColor = .listBackground
+    }
+
+    func configureTableView() {
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.backgroundColor = .listBackground
+
+        registerTableViewCells()
+
+        tableView.dataSource = self
+        tableView.delegate = self
+    }
+
+    func registerTableViewCells() {
+        for row in Row.allCases {
+            tableView.register(row.type.loadNib(), forCellReuseIdentifier: row.reuseIdentifier)
+        }
+    }
+}
+
+// MARK: - Navigation actions handling
+//
+private extension ProductShippingSettingsViewController {
+    @objc func completeUpdating() {
+        // TODO-1422: update shipping settings
+    }
+}
+
+// MARK: - Input changes handling
+//
+private extension ProductShippingSettingsViewController {
+    func handleWeightChange(weight: String?) {
+        self.weight = weight
+    }
+
+    func handleLengthChange(weight: String?) {
```

Comment:
I think you mean `length` for the parameter?

Reply:
yeaa my bad 🤦🏻‍♀️will fix!